### PR TITLE
feat: remove unnecessary and deprecated rules

### PR DESCRIPTION
**Pull request by frantic1048**
This PR resolves the step 2 of https://github.com/RightCapitalHQ/frontend-style-guide/issues/159

This PR depends on https://github.com/RightCapitalHQ/frontend-style-guide/pull/160

1. Reduced redundant rule overrides.
2. Removed useless rule configs.
3. Style related rules.
   1. Removed rules that conflict with Prettier (This make us no longer need to ship `eslint-config-prettier` as a dependency).
   2. Added test to check conflicts with Prettier.
   3. Migrated deprecated style rules to `@stylistic/js` counterparts.
4. Legacy rules (prior ESLint v8 )
   1. Removed/Migrated deprecated rules, except these ones(currently the have no ideal replacement, we may fix these rules later):
      - `no-buffer-constructor`
      - `no-return-await`
   2. Added test to check deprecated rules.

For detailed rule changes, please see https://github.com/RightCapitalHQ/frontend-style-guide/pull/166/files#diff-6a7b668335f2224bcd651ef1b8dd12c5e8ac3915e02590617e73529a6b5237ce

### CODE REVIEW

File: packages/eslint-config-base/src/rules/best-practices.ts [hunk 1]
```diff
@@ -16,26 +13,13 @@ const config: Linter.Config = {
...
-    ],
-
     // require return statements to either always or never specify values
     // https://eslint.org/docs/rules/consistent-return
     'consistent-return': 'error',
 
     // specify curly brace conventions for all control statements
     // https://eslint.org/docs/rules/curly
-    curly: ['error', 'multi-line'], // multiline
+    curly: ['error', 'all'],
```
Comment (liby) [inline packages/eslint-config-base/src/rules/best-practices.ts:22]: Will this result in detecting more errors in the existing code checks?

File: packages/eslint-config-typescript-react/src/rules/react-a11y.ts [hunk 2]
```diff
@@ -34,17 +29,6 @@ const config: Linter.Config = {
     // https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-has-content.md
     'jsx-a11y/anchor-has-content': ['error', { components: [] }],
 
-    // ensure <a> tags are valid
-    // https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/0745af376cdc8686d85a361ce36952b1fb1ccf6e/docs/rules/anchor-is-valid.md
-    'jsx-a11y/anchor-is-valid': [
```
Comment (liby) [inline packages/eslint-config-typescript-react/src/rules/react-a11y.ts:39]: Is it no longer needed?

File: packages/eslint-config-base/src/rules/best-practices.ts (hunk 1 above)
Comment (frantic1048) [reply packages/eslint-config-base/src/rules/best-practices.ts:22]: No. This rule remains the same prior to this PR.

This rule is previously overridden by us: https://github.com/RightCapitalHQ/frontend-style-guide/pull/166/files#diff-2c866a6ebba9015a6180c332af9b6f11eda5c1b1089cad4732a5625e11e65cf6L31-L32.

We always use `['error', 'all']` for this rule.

Additionaly, the `multi-line` setting conflicts with Prettier. Therefore, `eslint-config-prettier` disables the `curly` rule and leave it to the users manually specify a safe option: https://github.com/prettier/eslint-config-prettier?tab=readme-ov-file#curly

File: packages/eslint-config-base/src/rules/imports.ts [hunk 3]
```diff
@@ -100,73 +90,43 @@ const config: Linter.Config = {
...
-    // disallow namespace imports
-    // TODO: enable?
-    // https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-namespace.md
-    'import/no-namespace': 'off',
-
     // Ensure consistent use of file extension within the import path
     // https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/extensions.md
     'import/extensions': [
-      'error',
+      'warn',
```
[the remark this answers is not in the document]
Comment (frantic1048) [reply packages/eslint-config-base/src/rules/imports.ts:106]: This was overridden by us: https://github.com/RightCapitalHQ/frontend-style-guide/pull/166/files#diff-2c866a6ebba9015a6180c332af9b6f11eda5c1b1089cad4732a5625e11e65cf6L76-L85

However, I think we should use `error` here 🤔

**Review by liby — APPROVED** [review]

**Pull request merged**